ownCloud/Client/ClientRootViewController.swift

```diff
@@ -155,32 +166,28 @@ class ClientRootViewController: UITabBarController {
 
 		self.tabBar.applyThemeCollection(Theme.shared.activeCollection)
 
-		self.viewControllers = [filesNavigationController] as? [UIViewController]
-	}
-
-	func logoutBarButtonItem() -> UIBarButtonItem {
-		let barButton = UIBarButtonItem(title: "Disconnect".localized, style: .plain, target: self, action: #selector(logout(_:)))
-		barButton.accessibilityIdentifier = "disconnect-button"
-		return barButton
-	}
-
-	@objc func logout(_: Any) {
-		self.closeClient()
+        self.viewControllers = [filesNavigationController] as? [UIViewController]
 	}
 
 	func closeClient(completion: (() -> Void)? = nil) {
-		self.presentingViewController?.dismiss(animated: true, completion: completion)
+        self.navigationController?.setNavigationBarHidden(false, animated: false)
+        self.navigationController?.popViewController(animated: true, completion: {
+            
+        })
 	}
 
 	func coreReady() {
 		OnMainThread {
 			let queryViewController = ClientQueryViewController(core: self.core!, query: OCQuery(forPath: "/"))
-
-			queryViewController.navigationItem.leftBarButtonItem = self.logoutBarButtonItem()
-
-			self.filesNavigationController?.pushViewController(queryViewController, animated: false)
+            self.filesNavigationController?.setViewControllers([self.emptyViewController, queryViewController], animated: false)
```

Comment:
@hosy would at least explain in a comment, why you put emptyViewController onto navigation stack.

Reply:
Because we have nested UINavigationControllers (first one from ServerListTableViewController and each item UITabBarController needs it own UINavigationController), we have to fake the UINavigationController logic. Here we insert the emptyViewController, because in the UI should appear a "Back" button if the root of the queryViewController is shown. Therefore we put at first the emptyViewController inside and at the same time the queryViewController. Now, the back button is shown and if the users push the "Back" button the ServerListTableViewController is shown. This logic can be found in navigationController(_: UINavigationController, willShow: UIViewController, animated: Bool) below.

Reply:
@hosy as already pointed out, I am fine with the approach, I would just like a short comment in the code

Reply:
I added the comment above also to the code